packages/firestore/src/local/lru_garbage_collector.ts

```diff
@@ -267,23 +269,15 @@ export class LruScheduler implements GarbageCollectionScheduler {
     this.gcTask = this.asyncQueue.enqueueAfterDelay(
       TimerId.LruGarbageCollection,
       delay,
-      async () => {
+      () => {
         this.gcTask = null;
         this.hasRun = true;
-        try {
-          await localStore.collectGarbage(this.garbageCollector);
-        } catch (e) {
-          if (e.name === 'IndexedDbTransactionError') {
-            logDebug(
-              LOG_TAG,
-              'Ignoring IndexedDB error during garbage collection: ',
-              e
-            );
-          } else {
-            await ignoreIfPrimaryLeaseLoss(e);
-          }
-        }
-        await this.scheduleGC(localStore);
+        return executeWithIndexedDbRecovery(
```

Comment:
In particular, this change has fairly significantly harmed this method:

  * Why is the extra `.then(() => {})` required?
  * Going back to `then`/`catch` from `async`/`await` is a bummer.